test/e10.test.js

```diff
@@ -1,90 +1,213 @@
-    expect(handlePromise3(promiseArr).constructor.name).toEqual("Promise");
-  });
+    it("should get 2 failures if I pass in 2 failures and 2 successes", async () => {
+      const promises = [
+        wait(1000),
+        failAfterWith(500, "rejected"),
+        wait(2000),
+        failAfterWith(500, "rejected"),
+      ];
 
-  it("HandlePromise3 should use .allSettled() method", () => {
-    const promiseSpy = vitest.spyOn(Promise, "allSettled");
-    handlePromise3(promiseArr);
-    expect(promiseSpy).toBeCalled();
-  });
+      const result = getQuantityOfRejectedPromises(promises);
+      vitest.runAllTimers();
 
-  it("HandlePromise3 should return a resolved value of the array of all promises status and value/reason", async () => {
-    expect(await handlePromise3(promiseArr)).toEqual(sampleData);
+      expect(await result).toBe(2);
+    });
   });
-});
 
-describe("handlePromise4 variable test", () => {
-  it("handlePromise4 variable should exist", () => {
-    expect(handlePromise4).toBeInstanceOf(Function);
-  });
+  describe("getQuantityOfFulfilledPromises", () => {
+    it("should get me 1 resolved if I pass in three failing promises and 1 succeeding promise", async () => {
+      const promises = [
+        failAfterWith(1, "rejected"),
+        failAfterWith(1, "rejected"),
+        wait(1000),
+        failAfterWith(100, "rejected"),
+      ];
 
-  it("HandlePromise4 should return a promise", () => {
-    expect(handlePromise4(newPromiseArr).constructor.name).toEqual("Promise");
-  });
+      const result = getQuantityOfFulfilledPromises(promises);
+      vitest.runAllTimers();
+
+      expect(await result).toBe(1);
+    });
+
+    it("should get 3 resolved values if I pass in 3 successful promises", async () => {
+      const promises = [wait(1000), wait(2000), wait(4000)];
+
+      const result = getQuantityOfFulfilledPromises(promises);
+      vitest.runAllTimers();
 
-  it("HandlePromise4 should use .allSettled() method", () => {
-    const promiseSpy = vitest.spyOn(Promise, "race");
-    handlePromise4(promiseArr);
-    expect(promiseSpy).toBeCalled();
+      expect(await result).toBe(3);
+    });
   });
 
-  it("HandlePromise4 should return a resolved value of 'RESOLVED AGAIN'", async () => {
-    expect(await handlePromise4(newPromiseArr)).toEqual("RESOLVED AGAIN");
+  describe.concurrent("fetchAllCharactersByIds", () => {
+    beforeAll(() => {
+      vitest.useRealTimers();
+    });
+    afterAll(() => {
+      vitest.useFakeTimers();
+    });
+
+    it("[1,2], should return billy and mandy", async () => {
+      const result = fetchAllCharactersByIds([1, 2]);
+
+      expect(await result).toEqual([
+        allCharacters.find((c) => c.id === 1),
+        allCharacters.find((c) => c.id === 2),
+      ]);
+    });
+
+    it("[1,2,3] should return all characters", async () => {
+      const result = fetchAllCharactersByIds([1, 2, 3]);
+      expect(await result).toEqual(allCharacters);
+    });
+
+    it(
```

Comment:
whitespace. is there not a linter installed here?

Reply:
That passes the linter. You can either have that whitespace or not have it with prettier. Personally I prefer to have it because it makes it easier to flip around my code in vim with `{` and `}` which skip around by blocks of text

Reply:
For what it's worth I normally don't like setting super tight rules about whitespace on eslint, particularly because it get's annoying if you use prettier as well. So I just let prettier do what it does, and my preference is pretty much to always add space between code blocks, so I naturally just do that as often as possible.

Reply:
It's ugly. I'm pretty annoyign when it comes to whitespace. But whatever